mpas_analysis/shared/climatology/climatology.py

```diff
@@ -515,19 +530,105 @@ def update_start_end_year(ds, config, calendar):
     endYear = days_to_datetime(ds.Time.max().values,  calendar=calendar).year
     changed = False
     if startYear != requestedStartYear or endYear != requestedEndYear:
-        warn("climatology start and/or end year different from requested\n"
-             "requestd: {:04d}-{:04d}\n"
-             "actual:   {:04d}-{:04d}\n".format(requestedStartYear,
-                                                requestedEndYear,
-                                                startYear,
-                                                endYear))
+        warnings.warn("climatology start and/or end year different from requested\n"
+                      "requestd: {:04d}-{:04d}\n"
+                      "actual:   {:04d}-{:04d}\n".format(requestedStartYear, requestedEndYear, startYear, endYear))
         config.set('climatology', 'startYear', str(startYear))
         config.set('climatology', 'endYear', str(endYear))
         changed = True
 
     return changed, startYear, endYear
 
 
+def add_months_and_days_in_month(ds, calendar):
+    '''
+    Add ``months`` and ``daysInMonth`` as data arrays in ``ds``.  The number
+    of days in each month of ``ds`` is computed either using the ``startTime``
+    and ``endTime`` if available or assuming ``gregorian_noleap`` calendar and
+    ignoring leap years.
+
+    Parameters
+    ----------
+    ds : ``xarray.Dataset`` object
+        A data set with a ``Time`` coordinate expressed as days since
+        0001-01-01
+
+    calendar: ``{'gregorian', 'gregorian_noleap'}``
+        The name of one of the calendars supported by MPAS cores, used to
+        determine ``month`` from ``Time`` coordinate
+
+    Returns
+    -------
+    ds : ``xarray.Dataset`` object
+        The data set with ``month`` and ``daysInMonth`` data arrays added (if
+        not already present)
+
+    Authors
+    -------
+    Xylar Asay-Davis
+
+    Last Modified
+    -------------
+    03/29/2017
+    """    '''
+
+    ds = ds.copy()
+
+    if 'month' not in ds.data_vars and 'month' not in ds.coords:
+        if calendar is None:
+            raise ValueError('calendar must be provided if month coordinate is not in ds')
+        months = [date.month for date in days_to_datetime(ds.Time,
+                                                          calendar=calendar)]
+
+        ds['month'] = ('Time', months)
+
+    if 'daysInMonth' not in ds.data_vars:
+        if 'startTime' in ds.data_vars and 'endTime' in ds.data_vars:
+            ds['daysInMonth'] = ds.endTime - ds.startTime
+        else:
+            # TODO: support calendar
```

Comment:
I have elaborated.  I do not feel like writing special code for handling this case when the calendar is `gregorian`.  We would need to figure out whether each year is a leap year and whether the month is February, and change the number of days to 29 if so.  We don't know of any runs that will need this capability.  The `else` case here is to support backward compatibility to ACME alpha runs where we used `daysSinceStartOfSim` to compute the time coordinate (none of these runs use the `gregorian` calendar).  So I have added a warning but it doesn't seem like it's worth the effort of handling this case.  If you want to, feel free...